### DISCUSSION TITLE
Window scrolling does not work on Mac

### DISCUSSION THREAD

**Discussion by devns8** [wezterm/wezterm#6000] [Q&A]
### What Operating System(s) are you running on?

macOS

### WezTerm version

??

### Ask your question!

I have scoured the Github Issues and Discussion pages trying all sorts of configs.  My issue is, that on both, my iMac and Macbook, when I attempt to scroll the Wezterm window up to, for example, see the long output of a install / compilation / build, it only scrolls through my bash history commands, not the window itself.  

This happens when I use the mouse or trackpad on my macbook.  I've also tried to edit my configs and use the page-up / page-down buttons to no avail.  Love the terminal, but need to get this window scrolling working.  Below is my very basic config as it stands now. 

``` 
local wezterm = require 'wezterm'

 -- This will hold the configuration.
 local config = wezterm.config_builder()
 local act = wezterm.action

 -- This is where you actually apply your config choices

 -- For example, changing the color scheme:
 -- config.color_scheme = 'AdventureTime' --
 config.font_size = 15
 -- config.enable_scroll_bar = true--
 config.scrollback_lines = 1000

 --config.keys = {  { key = 'PageUp', mods = 'SHIFT', action = act.ScrollByPage(-1)  },  { key =
 'PageDown', mods = 'SHIFT', action = act.ScrollByPage(1)  }, }
 --
 --onfig.mouse_bindings = {
 -- {
 --   event = { Down = { streak = 1, button = { WheelUp = 1 } } },
 --   mods = 'NONE',
 --   action = act.ScrollByCurrentEventWheelDelta,
 -- },
 -- {
 --   event = { Down = { streak = 1, button = { WheelDown = 1 } } },
 --   mods = 'NONE',
 --   action = act.ScrollByCurrentEventWheelDelta,
 -- },

 -- and finally, return the configuration to wezterm --
 -- config.disable_default_mouse_bindings = true--
 return config
```

**Comment by Alien-lv426** [comment]
Works perfectly fine on my Mac with CTRL-SHIFT-U and CTRL-SHIFT-D bound to act.ScrollByPage. Maybe you need to disable the builtin key function first using wezterm.action.DisableDefaultAssignment ?

**Comment by AayushCS** [comment]
> ### What Operating System(s) are you running on?
> macOS
> 
> ### Which Wayland compositor or X11 Window manager(s) are you using?
> _No response_
> 
> ### WezTerm version
> ??
> 
> ### Ask your question!
> I have scoured the Github Issues and Discussion pages trying all sorts of configs. My issue is, that on both, my iMac and Macbook, when I attempt to scroll the Wezterm window up to, for example, see the long output of a install / compilation / build, it only scrolls through my bash history commands, not the window itself.
> 
> This happens when I use the mouse or trackpad on my macbook. I've also tried to edit my configs and use the page-up / page-down buttons to no avail. Love the terminal, but need to get this window scrolling working. Below is my very basic config as it stands now.
> 
> 
> local wezterm = require 'wezterm'
> 
>  -- This will hold the configuration.
>  local config = wezterm.config_builder()
>  local act = wezterm.action
> 
>  -- This is where you actually apply your config choices
> 
>  -- For example, changing the color scheme:
>  -- config.color_scheme = 'AdventureTime' --
>  config.font_size = 15
>  -- config.enable_scroll_bar = true--
>  config.scrollback_lines = 1000
> 
>  --config.keys = {  { key = 'PageUp', mods = 'SHIFT', action = act.ScrollByPage(-1)  },  { key =
>  'PageDown', mods = 'SHIFT', action = act.ScrollByPage(1)  }, }
>  --
>  --onfig.mouse_bindings = {
>  -- {
>  --   event = { Down = { streak = 1, button = { WheelUp = 1 } } },
>  --   mods = 'NONE',
>  --   action = act.ScrollByCurrentEventWheelDelta,
>  -- },
>  -- {
>  --   event = { Down = { streak = 1, button = { WheelDown = 1 } } },
>  --   mods = 'NONE',
>  --   action = act.ScrollByCurrentEventWheelDelta,
>  -- },
> 
>  -- and finally, return the configuration to wezterm --
>  -- config.disable_default_mouse_bindings = true--
>  return config
>


probably because you are opening a tmux window and you do not have the following line in your configuration file of tmux

```bash
set -g mouse on
```

add this to your ~/.tmux.conf file and restart your terminal and scrolling should work fine.
